indico/modules/events/registration/util.py

```diff
@@ -0,0 +1,43 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from indico.modules.events.registration.models.items import RegistrationFormSection
+from indico.web.forms.base import IndicoForm
+
+
+def get_event_section_data(regform):
+    if not regform.event.has_feature('registration'):
+        return []
+    sections = (RegistrationFormSection.find(registration_form=regform, is_deleted=False)
+                                       .order_by(RegistrationFormSection.position))
+    return [s.view_data for s in sections]
+
+
+def make_registration_form(regform):
+    """Creates dynamically a WTForm based on registration form fields"""
+
+    form_class = type(b'RegistrationFormWTForm', (IndicoForm,), {})
+    form_items = [field for field in regform.form_items if field.parent_id and not field.is_deleted
+                  and field.is_enabled and not field.parent.is_deleted and field.parent.is_enabled]
+    for form_item in form_items:
+        field_impl = form_item.wtf_field
+        if field_impl is None:
+            continue
+        name = '*genfield*{}-{}'.format(form_item.parent_id, form_item.id)
```

Comment:
:volcano: :cancer: :poop: :bomb: 

Let's get rid of this horrible naming scheme once and for all!